### PR TITLE
[Quantization][Refactor] Make profile include only min/max

### PR DESCRIPTION
**Pull request by mciprian13**
**Summary**

Until this point the profile generated after the profiling phase included the quantization parameters **scale** and **offset** which were computed for a given quantization configuration (schema, precision, precisionBias). This approach was extremely error prone (multiple times I had this problem) because in order for the flow to be consistent, same configuration had to be provided during the profiling (e.g. while using image-classifier or model-profiler) versus during the quantization (e.g. while using the model-compiler).

With this PR the profile includes only the profiling specific information (**min** and **max**) while the actual quantization parameters are computed during the quantization phase. This has the following benefits:
- the profiling phase becomes totally independent on the quantization configuration (schema, precision, precisionBias) thus making the usage of the front-end tools less error prone (for model-profiler or image-classifier no quantization parameters are required)
- the information in the **profile.yml** is much more readable in terms of the tensor dynamic range (which I think was the purpose of using the YAML format in the first place) because just having the **scale** and **offset** was not indicative on the tensor dynamic range because:
  - the information itself was not sufficient without the data precision (given offset & scale mean totally different things for int8 vs int32)
  - some information was lost because for (e.g.) Symmetric quantization schema the information about the original (min,max) pair was lost due to symmetrization
- the profiling information is closer to where the actual quantization is performed, allowing more elaborate and precise quantization schemes (joint operand quantization, etc)

The functionality of serializing the tensor quantization parameters (scale & offset) is still there and we might use it for debugging (although the scales and offset are readable in the dumped graph).

**Fixes**
#2600
#2599 (No longer required)

**Documentation**
Updated AOT.md.

**Test Plan**
The quantization unit tests are passing.
Added new unit test for the new **getNodeValueByName()** utility function.
I manually tested the LeNet MNIST model and verified the results of the quantized bundle are exactly the same (bit exact) before and after this change. This is possible because the serialization of the min/max preserves the floating-point bit-exactness (just as it did for the floating-point scale).

**Next**
We should also include in the profile the histogram (which is already computed) and use some smarter quantization (outlier removal, KL divergence minimization, etc).

### CODE REVIEW

**Comment by mciprian13** [comment]
@jackm321 Can we have this reviewed? Thanks!

File: lib/Quantization/Quantization.cpp [hunk 1]
```diff
@@ -702,15 +701,15 @@ class FunctionQuantizer : public FunctionConverter {
...
+  /// Creates a function quantizer for \p F using the profiling
+  /// parameters defined by \p profilingInfos and target quantization
   /// precision defined by \p quantizationPrecision.
   /// \p B and \p doNotQuantizeKinds are used to check which nodes shouldn't be
   /// converted. \p assertAllNodesQuantized is used as a debugging tool; if
   /// true then if the backend does not support a node as quantized for the
   /// given \p quantizationPrecision then the program will exit with an error.
   FunctionQuantizer(Function &F, const Backend &B, quantization::Schema schema,
-                    llvm::ArrayRef<NodeQuantizationInfo> quantizationInfos,
+                    const std::vector<NodeProfilingInfo> &profilingInfos,
```
Comment (jackm321) [inline lib/Quantization/Quantization.cpp]: Why the change away from ArrayRef? Can generateNodeQuantizationInfos just accept ArrayRef<NodeProfilingInfo>?

File: include/glow/Quantization/Quantization.h [hunk 2]
```diff
@@ -30,16 +30,30 @@ class Backend;
...
+/// the function \ref generateNodeProfilingInfos. The quantization schema is
+/// specified by \p schema and the target precision for quantization is given
+/// by \p quantizationPrecision for all the operands except the bias operands
+/// (used in nodes like Convolution and FullyConnected) for which the target
+/// precision is given by \p quantizationPrecisionBias. The map \p loweredMap
+/// is the lowering map obtained during the quantization phase and is used
+/// to find lowering patterns for the bias operands.
 std::vector<NodeQuantizationInfo> generateNodeQuantizationInfos(
-    PlaceholderBindings &bindings, const Function *F,
+    const std::vector<NodeProfilingInfo> &profilingInfos, Function *F,
```
Comment (jackm321) [inline include/glow/Quantization/Quantization.h]: ArrayRef?

File: lib/Quantization/Quantization.cpp (hunk 1 above)
Comment (mciprian13) [reply lib/Quantization/Quantization.cpp]: I changed it back to llvm::ArrayRef.

File: include/glow/Quantization/Quantization.h (hunk 2 above)
Comment (mciprian13) [reply include/glow/Quantization/Quantization.h]: Changed it to llvm::ArrayRef.

**Comment by mciprian13** [comment]
@jackm321 Can we have this landed? Thanks!

**Comment by mciprian13** [comment]
Side question: why the preference for llvm::StringRef and llvm::ArrayRef instead of std::string and std::vector? What are the benefits?

**Comment by jackm321** [comment]
@mciprian13 llvm::StringRef and llvm::ArrayRef are preferred just because they don't mandate the underlying container of the argument

**Comment by mciprian13** [comment]
@jackm321 Thanks for the explanation. Can we have this landed?

**Comment by mciprian13** [comment]
@jfix71 Can we have this landed? I`m working on other quantization refactorings and I need this landed. Thanks!
